### ISSUE TITLE
Race condition in slider value validation

### ISSUE THREAD

**Issue by ssunkara1** [ipython/ipython#6814]
```python
from IPython.html.widgets import *
from IPython.display import display
a = IntSlider(description='Test')
display(a)
```

```python
a.min = 20
a.value = 50
```

```python
print a.value
## displays 20 instead of 50. Although the slider visually is at 50
```
What is happening is, the value is set from the Javascript side on change of min. This is happening not necessarily before the value is set on the python side in the next statement. 

I was able to avoid the issue by setting the value on the min_changed function on the python side and not updating it from JS.

```python
    def _handle_min_changed(self, name, old, new):
        """Make sure the max is always >= the min and change value if needed."""
        if new > self.max:
            raise ValueError("setting min > max")
        if self.value < self.min:
            self.value = self.min
```

What do you guys think will be the best way to deal with this?

Thanks,
Srinivas

**Comment by ellisonbg** [comment]
What is the status of this? @jdfreder should be fixed for 3.0?

**Comment by minrk** [comment]
Anyone object to bumping this past 3.0? I don't think it's critical.

**Comment by SylvainCorlay** [comment]
It is an easy fix, I can open a PR tomorrow or Sunday if you like.

**Comment by ellisonbg** [comment]
Thanks! If there are other widget bugs you think should be fixed before
3.0, let us know and we can try to figure out how to get them done.

On Fri, Jan 23, 2015 at 5:18 PM, Sylvain Corlay <notifications@github.com>
wrote:

> It is an easy fix, I can open a PR tomorrow or Sunday if you like.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipython/issues/6814#issuecomment-71293167>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by SylvainCorlay** [comment]
Solved with #7602.
